Makefile

```diff
@@ -1,10 +1,35 @@
 ST2_VERSION ?= 3.4dev
 DOCKER_TAG ?= ${ST2_VERSION}
+RELEASE_TAG_REGEX := [^dev]$$
 SHELL := /bin/bash
 
+# supported values of the TAG_UPDATE_FLAG
+# 0 = no additional tags to be set
+# 1 = add the major.minor tag
+# 2 = add the tags major and major.minor
+# 3 = add the tags major, major.minor and latest
+TAG_UPDATE_FLAG := $(shell ./determine_needed_tags.sh st2 ${ST2_VERSION})
+
+ifneq ($(shell echo ${ST2_VERSION} | grep -E "${RELEASE_TAG_REGEX}"), )
+	RELEASE_VERSION := true
+	MAJOR := $(word 1, $(subst ., ,${ST2_VERSION}))
+	MINOR := $(word 2, $(subst ., ,${ST2_VERSION}))
+	PATCH := $(word 3, $(subst ., ,${ST2_VERSION}))
+else
+	RELEASE_VERSION := false
+endif
+
 # Build all required images (st2 base image plus st2 components)
 .PHONY: build
 build:
+ifeq ($(RELEASE_VERSION), true)
+ifneq ($(shell echo "${TAG_UPDATE_FLAG}" | grep -E "Error:"),)
+	@echo -e "Failed to identify the tags to be set."
+	@echo -e "\033[31mNo images were tagged due to an error when determining the correct tags: ${TAG_UPDATE_FLAG}\033[0m"
+	exit 1
+endif
+endif
```

Comment:
Looks like the same block repeated in both `build` and `push`.

I think it would be great to create a dedicated makefile target to avoid code duplication.

Reply:
Good point! Done.